juju/juju
# Get CharmProfilingInfo from state

### Checklist


----

## Description of change

Bug 1856832 has shown that when applying lxd profiles to existing containers, it was possible that the actual profile was empty.  This could only happen when there was more than 1 unit of the given charm and was not guaranteed to happen.  This was not seen on newer, faster machines.  

The lxd profile contents was retrieved from the model cache.  There is an outstanding issue to find of why the profile was applied with empty contents, there are checks to prevent that.  

This fix changes CharmProfileInfo to get data from state rather than the model cache.

## QA steps

Note: these two charms will have missing relations and that is okay for the purposes of this test, as the issue is seen or not when the subordinate (neutron-openvswitch) is added.
```sh
juju bootstrap lxd
juju deploy nova-compute -n 8 --to lxd,lxd,lxd,lxd,lxd,lxd,lxd,lxd
juju deploy cs:neutron-openvswitch-271
juju add-relation nova-compute neutron-openvswitch
```
Check that the lxd profile 'juju-default-neutron-openvswitch-271' applied to each container is not empty.
```sh
juju ssh 0 -- sudo  lxc profile show juju-default-neutron-openvswitch-271
```
 
## Bug reference

https://bugs.launchpad.net/juju/+bug/1856832
